internal/gengapic/client_init.go

```diff
@@ -73,80 +74,63 @@ func (g *generator) clientOptions(serv *descriptor.ServiceDescriptorProto, servN
 
 	// defaultCallOptions
 	{
-		type methodCode struct {
-			method string
-			codes  []code.Code
-		}
-
-		var idempotent []string
-		var overrideRetry []methodCode
-
-		for _, m := range serv.GetMethod() {
-			if m.GetOptions() == nil {
-				continue
-			}
-
-			eHttp, err := proto.GetExtension(m.GetOptions(), annotations.E_Http)
-			if err == proto.ErrMissingExtension {
-				continue
-			}
-			if err != nil {
-				return errors.E(err, "cannot read HTTP annotation")
-			}
-			// Generator spec mandates we bucket methods into idempotent and non-idempotent for retries, unless there is an override.
-			// TODO(pongad): implement the override.
-			if _, ok := eHttp.(*annotations.HttpRule).Pattern.(*annotations.HttpRule_Get); ok {
-				idempotent = append(idempotent, m.GetName())
+		policies := map[string]*conf.MethodConfig_RetryPolicy{}
+
+		// gather retry policies from gRPC ServiceConfig
+		if g.grpcConf != nil {
```

Comment:
I think the block below is a little uncomfortably far to the right. Maybe something like this might help?
```
var methCfgs []MethodConfig
if g.grpcConf != nil {
  methCfgs = g.grpcConf.GetMethodConfig()
}
for ...
```

Reply:
Good call. Done!